bin/git-forgit

```diff
@@ -386,66 +409,79 @@ _forgit_cherry_pick_from_branch() {
+        _forgit_extract_sha)
     if [[ -n "$target_commit" ]]; then
         prev_commit=$(_forgit_previous_commit "$target_commit")
-
-        $git_rebase "$prev_commit"
+        # shellcheck disable=SC2086 # split FORGIT_REBASE_GIT_OPTS on purpose
+        git rebase -i $FORGIT_REBASE_GIT_OPTS "$prev_commit"
     fi
 }
 
+_forgit_file_preview() {
+    echo "$1" | _forgit_extract_sha | xargs -I% git show --color=always % -- ${2:+"$2"} | $_forgit_show_pager
+}
+
 _forgit_fixup() {
     _forgit_inside_work_tree || return 1
     git diff --cached --quiet && echo 'Nothing to fixup: there are no staged changes.' && return 1
-    local git_fixup cmd preview opts graph files target_commit prev_commit
-    git_fixup="git commit --fixup $FORGIT_FIXUP_GIT_OPTS"
-    graph=--graph
-    [[ $FORGIT_LOG_GRAPH_ENABLE == false ]] && graph=
-    cmd="git log $graph --color=always --format='$_forgit_log_format' $* $_forgit_emojify"
-    files=$(sed -nE 's/.* -- (.*)/\1/p' <<< "$*")
-    preview="echo {} | $_forgit_extract_sha | xargs -I% git show --color=always % -- $files | $_forgit_show_pager"
+    local opts graph files target_commit prev_commit
+    graph=()
+    [[ $_forgit_log_graph_enable == true ]] && graph=(--graph)
+    IFS=$'\n'
+    #shellcheck disable=2207
+    files=($(sed -nE 's/.* -- (.*)/\1/p' <<< "$*"))
+    ${old_IFS+"false"} && unset IFS || IFS="$old_IFS"
     opts="
         $FORGIT_FZF_DEFAULT_OPTS
         +s +m --tiebreak=index
-        --bind=\"ctrl-y:execute-silent(echo {} | $_forgit_extract_sha | ${FORGIT_COPY_CMD:-pbcopy})\"
-        --preview=\"$preview\"
+        --bind=\"ctrl-y:execute-silent($FORGIT yank_sha {})\"
+        --preview=\"$FORGIT file_preview {} \"${files[*]}\"\"
         $FORGIT_FIXUP_FZF_OPTS
     "
-    target_commit=$(eval "$cmd" | FZF_DEFAULT_OPTS="$opts" fzf | eval "$_forgit_extract_sha")
-    if [[ -n "$target_commit" ]] && $git_fixup "$target_commit"; then
+    target_commit=$(
+        git log "${graph[@]}" --color=always --format="$_forgit_log_format" "$@" ${_forgit_emojify:+$_forgit_emojify} |
+        FZF_DEFAULT_OPTS="$opts" fzf |
+        _forgit_extract_sha)
+    # shellcheck disable=SC2086 # split FORGIT_FIXUP_GIT_OPTS on purpose
+    if [[ -n "$target_commit" ]] && git commit --fixup $FORGIT_FIXUP_GIT_OPTS "$target_commit"; then
         prev_commit=$(_forgit_previous_commit "$target_commit")
         # rebase will fail if there are unstaged changes so --autostash is needed to temporarily stash them
         # GIT_SEQUENCE_EDITOR=: is needed to skip the editor
         GIT_SEQUENCE_EDITOR=: git rebase --autostash -i --autosquash "$prev_commit"
     fi
+}
 
+_forgit_checkout_file_preview() {
+    git diff --color=always -- "$1" | $_forgit_diff_pager
 }
 
 # git checkout-file selector
 _forgit_checkout_file() {
     _forgit_inside_work_tree || return 1
-    local git_checkout cmd files opts
+    local git_checkout files opts
     git_checkout="git checkout $FORGIT_CHECKOUT_FILE_GIT_OPTS"
```

Comment:
@sandr01d In `_forgit_checkout_file` we still have a deferred code line for the git command, which we're using for returning early as well as the action after the fzf dialog. This is common practice in other forgit functions as well. Do we want to keep this? IMO it makes sense, because it reduces code duplication. It also solves the problem with the options variable, so for consistency we could do this in every function and wouldn't need to disable the shellcheck.
Cleaner, however, would be defining a function for it. What do you think?

Reply:
I would prefer to use functions here a as well.  When we parse the options environment variables into an array as you mentioned earlier we don't even have to disable the shellcheck and I find functions more readable and less error prone to deferred execution.

Reply:
Alright, then let's go with functions here as well!